### PR TITLE
Replace `KILL` signal with `TERM` for `:restart` option

### LINKED ISSUE #106: --restart flag support process kill with custom signal?
I'm running into an issue using `filewatcher` to run and restart because I require the child process to perform some cleanup before being force killed with KILL 9 signal.

With my limited understanding of the intricacies of this gem, it seems like it should be pretty straight forward to support any value instead of `9` here:

https://github.com/filewatcher/filewatcher/blob/575e1d5a0dad2f079684ec97e4b084ea9f1e71ab/bin/filewatcher#L56

I'd be happy to implement this feature. Please let me know if I'm missing anything and if you'd accept a PR

### PR DESCRIPTION
**Pull request by AlexWayfer**
I don't see needing for `KILL` signal, also we're waiting for termination.

Resolve #106

### CODE REVIEW

File: bin/filewatcher [hunk 1]
```diff
@@ -54,7 +54,7 @@ def restart(pid, env, cmd)
   begin
     raise Errno::ESRCH unless pid
 
-    Process.kill(9, pid)
+    Process.kill('TERM', pid)
```
Comment (TSMMark) [inline bin/filewatcher:57]: Seems like the term signal could be trapped and the process may never exit right? Do we need to have a wait for graceful exit timeout and then kill -9 if it takes too long to die?

Comment (AlexWayfer) [reply bin/filewatcher:57]: > Seems like the term signal could be trapped and the process may never exit right?

Yes.

Any signal can be trapped except `KILL`.

> Do we need to have a wait for graceful exit timeout and then kill -9 if it takes too long to die?

I thought about it. But I don't know how long. Option? With which default? I guess it should be resolved after problematic cases.

Comment (TSMMark) [reply bin/filewatcher:57]: > Option?

Sure. I was originally proposing the kill signal as a new option as well. 

> With which default?

My first thought is that a few seconds should be fine. Hopefully most processes would exit properly and it won't come up

Comment (AlexWayfer) [reply bin/filewatcher:57]: I don't want to implement something in what we don't need guaranteed.

If you want to test this for your case — let's do it and find out any problems, if not — I'm going to wait new problems, if they will be here at all (I'm not sure about this).

**Comment by AlexWayfer** [comment]
Closing in favor of #132, where the actual option for signal.

**Pull request closed without merging**